### ISSUE TITLE
Directive allows spaces after number when using Chrome 43.0.2357.132

### ISSUE THREAD

**Issue by TwistyPop** [AlphaGit/ng-pattern-restrict#26]
Using the following pattern ^\d*$ to limit the entry to numbers only, this version of chrome allows the spaces until another invalid value is entered (such as an alpha). It is as if the routine you designed to select the invalid input and replace it with nothing (reverting to the old value) doesn't select trailing spaces. It still works in Firefox 35.0.1 and IE9. Could there be an issue with the selection.modify?

**Comment by AlphaGit** [comment]
Interesting... Is this with a input=type text? (I'd really appreciate a snippet of code to check it out.)

It is possible that it is an issue with the selection.modify, yes. However, in the way that you described the issue, I sounds more like the kind of issues I was facing with non-text inputs (like number, or email, or telephone). However, I'll be glad to take a look at your case and determine what's the cause.

**Comment by TwistyPop** [comment]
Thank you for the quick response! You are correct it seems to only be happening with type=number. Here is a snippet of the HTML that was used: 

    <div class="col-xs-8">
    <input type="number" id="days" ng-model="leadDay.lead" class="form-control input-sm" ng-pattern-restrict pattern="^[0-9]*$" />
    </div>

**Comment by AlphaGit** [comment]
Thanks for providing that. I will be taking a look at it in the following days, sorry I don't have much more availability to jump onto this.

I was hoping you told me it was an input type=text, since as you know, there are special restrictions on other types of inputs. In a previous ticket I [struggled with Chrome not providing the valid input](https://github.com/AlphaGit/ng-pattern-restrict/issues/2) and Opera [doing funky stuff of its own](https://github.com/AlphaGit/ng-pattern-restrict/issues/12).

With that said, we're pretty much limited into what the browser can tell us about the input: if it's valid, and its value. Chrome will not provide any value if the input is invalid, and may be trimming spaces for the numeric value. If that's the case, then I need to investigate a little bit further to see what can be done, but not much from the last time I searched.

Without relying on the input value, we can perform tricks to get the actual contents (like the selection methods used in the code), but this gets more difficult as non-text inputs will disallow us from using the selection API.

*Bottom line:* it does not look like there's much that we can do, but I'll take a look in the next few days and get back to you. Thanks!

**Comment by AlphaGit** [comment]
Hi! I've tested on Chrome 43, 44 and 45. Chrome 43 displays this behavior but 44 and 45 do not.

Furthermore, Chrome 43 will report the value property of a number input as "12" even if the entered text is "12   ", "1   2", "12", etc. 

I looked around but I was not able to find any bug report on this behavior. Anyway, it is very likely that it is there, somewhere, and I just didn't find it.

Being that the problem is solved now, I'll close this. Thanks!
